### ISSUE TITLE
Java: semmle-extractor-options setting classpath prevents usage on Windows

### ISSUE THREAD

**Issue by Marcono1234** [github/codeql#5346]
(Might affect extractors other than the Java one as well)

### Version
CodeQL CLI 2.4.5

### Description
Tests which use the `semmle-extractor-options` and set `--javac-args -cp` cannot be run on Windows because CodeQL CLI apparently uses `:` as PATH separator on all OS', including Windows, which causes issues because it strips off the drive letter (e.g. `C:`) and treats it as separate PATH location.

For example trying to run [library-tests/frameworks/apache-http](https://github.com/github/codeql/tree/main/java/ql/test/library-tests/frameworks/apache-http) from the `C:` drive on Windows from VSCode causes:
> Could not extract a dataset in C:\path\to\codeql\java\ql\test\library-tests\frameworks\apache-http: Path not found: C

Interestingly when the `-cp` specifies multiple paths separated using `:`, this separator seems to be converted to `;` based on the extractor log (assuming that the paths themselves do not contain a `:`). Therefore the current logic might be:
1. Expand `${testdir}`
2. Replace `:` with `;`

So maybe this could be solved by switching the steps to prevent `:` in the expanded `${testdir}` to be replaced (which is apparently causing this issue).

### Reproduction steps
1. Clone https://github.com/github/codeql
2. Open the console in the top level `codeql` folder
3. Run the following command
    ```
    codeql.exe test run --search-path=.\java\ql .\java\ql\test\library-tests\frameworks\apache-http
    ```
    :x: It fails with the following error (assuming that your drive letter is `C:`)
    ```
    Could not extract a dataset in C:\my projects\codeql\java\ql\test\library-tests\frameworks\apache-http: Path not found: C
    Path not found: C
    ```

**Comment by Marcono1234** [comment]
@hmakholm should this be moved to github/codeql-cli-binaries?

**Comment by hmakholm** [comment]
Hmm, the CLI _itself_ will definitely use `;` rather than `:` as a path separator on Windows. Whatever goes on here must be happening in the Java extractor, so this is the right repo. Perhaps @github/codeql-java can weigh in here?

**Comment by Marcono1234** [comment]
This is still happening with CodeQL 2.8.1

I would claim the issue is in `tools/codeql.jar` and that my original theory is correct, but I cannot provide an proof. Decompilation of CodeQL is prohibited by the license.

As workaround Windows users can edit the `options` file and:
1. Replace `-cp` with `--class-path` (because for that CodeQL does not perform any replacement)
2. Replace path separators `:` with the Windows path separator `;`

For the apache-http example mentioned above the modified `options` file would look like this:
```
//semmle-extractor-options: --javac-args --class-path ${testdir}/../../../stubs/apache-http-4.4.13;${testdir}/../../../stubs/apache-http-5
```

**Comment by smowton** [comment]
Fix committed, should work as hoped in the next release.

**Comment by Marcono1234** [comment]
Thanks, seems to be fixed in 2.8.2!
